Spanner/src/Instance.php

```diff
@@ -216,7 +216,18 @@ public function info(array $options = [])
     public function exists(array $options = [])
     {
         try {
-            $this->reload($options = []);
+            $options += [
+                'fieldMask' => ['name'],
+            ];
+
+            if ($this->info) {
+                $this->connection->getInstance($options + [
+                    'name' => $this->name,
+                    'projectId' => $this->projectId,
+                ]);
+            } else {
+                $this->reload($options);
```

Comment:
This will cache a subset of data. Put lines 219-221 inside the `if ($this->info) {` block. If there's no cache, don't set the field mask and let it work as normal, returning the entire resource.

Reply:
> Please also add a system test demonstrating the expected behavior for `exists()` with cached resource and without.

Not quite sure what _system_ test is needed for. As I see, we need to make sure that:
- `info` field is filled on cold (no cached data) `exists()` call;
- `info` field is not modified on warm (has cached data) `exists()` call.

These checks can be made with unit tests using mocked connection. Could you please explain what should system test do? Am I missing something?

Reply:
No, that's fair. Unit tests should be enough. Can you make the other change though?

Reply:
No problem, I'll fix it.